src/app/components/ContentExcerpt.tsx

```diff
@@ -0,0 +1,59 @@
+import { Highlight } from '@openstax/highlighter/dist/api';
+import { HTMLAnchorElement } from '@openstax/types/lib.dom';
+import React from 'react';
+import { useSelector } from 'react-redux';
+import styled from 'styled-components/macro';
+import { content } from '../content/routes';
+import { book } from '../content/selectors';
+import { BookWithOSWebData } from '../content/types';
+import { fromRelativeUrl, getUrlParamForPageId } from '../content/utils/urlUtils';
+import { bodyCopyRegularStyle } from './Typography';
+
+// tslint:disable-next-line:variable-name
+export const HighlightContent = styled.div`
+  ${bodyCopyRegularStyle}
+  overflow: auto;
+
+  * {
+    overflow: initial;
+  }
+`;
+
+interface Props {
+  highlight: Highlight;
+  content: any;
+  className: string;
+}
+
+// tslint:disable-next-line:variable-name
+const ContentExcerpt = (props: Props) => {
+  const currentBook = useSelector(book);
+
+  if (!currentBook) {
+    return null;
+  }
+
+  const absolutePath = content.getUrl({
+    book: {
+      slug: (currentBook as BookWithOSWebData).slug,
+    },
+    page: getUrlParamForPageId(currentBook, props.highlight.sourceId),
+  });
+
+  const parser = new DOMParser();
+  const domNode = parser.parseFromString(props.content, 'text/html');
+  domNode.querySelectorAll('a').forEach(
+    (a: HTMLAnchorElement) => {
+      const currentHref = a.getAttribute('href') || '';
```

Comment:
these need to be filtered to only relative urls

Reply:
to know if it's a relative url I need to get the `href` first, right? Do I need to filter before grabbing the href?

Reply:
you could theoretically do it in the selector, or you could filter the set after, but you're not doing any filtering right now so you gotta add that somewhere